core/variant/variant_parser.cpp

```diff
@@ -1300,7 +1300,7 @@ Error VariantParser::parse_value(Token &token, Variant &value, Stream *p_stream,
 			}
 
 			value = arr;
-		} else if (id == "PackedVector3Array" || id == "PoolVector3Array" || id == "Vector3Array") {
+		} else if (id == "PackedVector3Array" || id == "PoolVector3Array" || id == "Vector3Array" || id == "Vector4Array") {
```

Comment:
This looks wrong. A `Vector4Array` is not equivalent to a `Vector3Array`. I guess you meant to add this to the next `if` with `ColorArray`.

Also, it's missing id `PackedVector4Array` which is the one that we really want, because I'm pretty sure the others are just aliases for backwards compatibility.

Reply:
Thank you for finding the bug. Will probably work on this as soon as I can, but we landed a large pr recently.

Reply:
If anyone wants to help out on this, feel free to send me a branch with the changes. I'm a bit slow right now.

Reply:
```cpp
		} else if (id == "PackedVector4Array" || id == "PoolVector4Array" || id == "Vector4Array") {
			Vector<real_t> args;
			Error err = _parse_construct<real_t>(p_stream, args, line, r_err_str);
			if (err) {
				return err;
			}
			Vector<Vector4> arr;
			{
				int len = args.size() / 4;
				arr.resize(len);
				Vector4 *w = arr.ptrw();
				for (int i = 0; i < len; i++) {
					w[i] = Vector4(args[i * 4 + 0], args[i * 4 + 1], args[i * 4 + 2], args[i * 4 + 3]);
				}
			}
			value = arr;
```
Add below